server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+        .add(new InstanceType("db.t2.2xlarge", 8, 32768))
+        .add(new InstanceType("db.m4.4xlarge", 16, 65536))
+        .add(new InstanceType("db.m4.10xlarge", 40, 163840))
+        .add(new InstanceType("db.m4.16xlarge", 64, 262144))
+        .build();
+
+    /**
+     This method requests the AWS server for ImageIds with filters which are filled based on
+     the values of the OsCapability. The image with the latest creation date is picked and its imageId returned.
+
+     @param osCapability The OsCapability to map.
+     @return A String that contains a valid ImageId that can be added to the properties of an ec2.
+     */
+    public String mapOsCapabilityToImageId(OsCapability osCapability) {
+        AmazonEC2 ec2 = AmazonEC2ClientBuilder.standard()
+            .withRegion(Regions.US_WEST_2) //TODO get this from user
+            .build();
+        //need to set these, owner are self and amazon
+        DescribeImagesRequest describeImagesRequest = new DescribeImagesRequest()
+            .withFilters(
+                new Filter("virtualization-type").withValues("hvm"),
+                new Filter("root-device-type").withValues("ebs"))
+            .withOwners("099720109477"); //this is the ownerId of amazon itself
+        if (osCapability.getType().isPresent() && osCapability.getType().get().equals(OsCapability.Type.WINDOWS)) {
+            describeImagesRequest.withFilters(new Filter("platform").withValues("windows"));
+        }
+        if (osCapability.getDistribution().isPresent()) {
+            if (osCapability.getDistribution().get().equals(OsCapability.Distribution.UBUNTU)) {
+                // */ubuntu/images/* gets better results than plain *ubuntu*
+                describeImagesRequest.withFilters(new Filter("name").withValues("*ubuntu/images/*"));
+            } else {
+                //just search for the string
+                describeImagesRequest.withFilters(new Filter("name").withValues("*" + osCapability.getDistribution()
+                    .toString() + "*"));
+            }
+        }
+        if (osCapability.getVersion().isPresent()) {
+            describeImagesRequest.withFilters(new Filter("name").withValues("*" + osCapability.getVersion().get() +
+                "*"));
+        }
+        if (osCapability.getArchitecture().isPresent()) {
+            if (osCapability.getArchitecture().get().equals(OsCapability.Architecture.x86_64)) {
+                describeImagesRequest.withFilters(new Filter("architecture").withValues("x86_64"));
+            } else if (osCapability.getArchitecture().get().equals(OsCapability.Architecture.x86_32)) {
+                describeImagesRequest.withFilters(new Filter("architecture").withValues("i386"));
+            }
+        } else {
+            //defaulting to 64 bit architecture
+            describeImagesRequest.withFilters(new Filter("architecture").withValues("x86_64"));
+        }
+        String imageId;
```

Comment:
You could probably extract this part until the return statement and return something like mapImageId(...) instead.